### PR TITLE
Tests: Add vagrant.boxes gradle property

### PR DESCRIPTION
**Pull request by rjernst**
This change makes specifying which boxes to run vagrant tests on a
little easier. Previously there were two tasks, checkPackages and
checkPackagesAllDistros. With this change, there is a single
packagingTest task. The boxes to run on are specified using the
gradle property vagrant.boxes, which can be easily specified on the
command line, or in a gradle properties file. There are also two
alias names, 'sample' for a yum and apt box, and 'all' for all boxes.

### CODE REVIEW

File: qa/vagrant/build.gradle [hunk 1]
```diff
@@ -127,21 +131,22 @@ task checkVagrantVersion(type: Exec) {
...
+              "    test all available boxes. The available boxes are: \n    ${availableBoxes}"
 }
 
 // Each box gets it own set of tasks
-boxes.each { box ->
-  String boxTask = taskifyBoxName box
-  task "vagrantUp${boxTask}"(type: VagrantCommandTask) {
-    group 'Vagrant'
-    description "Startup a vagrant VM running ${box}"
+for (String box : boxes) {
```
Comment (nik9000) [inline qa/vagrant/build.gradle]: Maybe this should be done against availableBoxes and you should conditionally add the dependsOn to vagrantSmokeTest and packagingTest? That way `gradle clean` will try to halt all the boxes.

**Comment by nik9000** [comment]
Mostly LGTM - left a question about whether we should generate the tasks always and use the list only for generating dependencies.

File: qa/vagrant/build.gradle [hunk 2]
```diff
@@ -127,21 +131,22 @@ task checkVagrantVersion(type: Exec) {
...
 
-task vagrantSmokeTestAllDistros {
-  group 'Vagrant'
-  description 'Smoke test all distros from the Vagrantfile'
+task packagingTest {
+  group 'Verification'
+  description "Tests yum/apt packages using vagrant and bats.\n" +
+              "    Specify the vagrant boxes to test using the gradle property 'vagrant.boxes'.\n" +
+              "    'sample' can be used to test a single yum and apt box. 'all' can be used to\n" +
+              "    test all available boxes. The available boxes are: \n    ${availableBoxes}"
```
Comment (nik9000) [inline qa/vagrant/build.gradle]: Maybe put the `    ${availableBoxes}` on the next line?

**Comment by rjernst** [comment]
Thanks @nik9000 I pushed commits to address your comments.

**Comment by nik9000** [comment]
LGTM

**Comment by dakrone** [comment]
@rjernst I think TESTING.asciidoc also needs to be updated as it points to the old (removed) gradle tasks now?

**Comment by rjernst** [comment]
@dakrone Good catch, I updated the docs in d1cfe0e.

**Pull request merged**